File: Task3/Golang/subtask4.go
package main

import (
    "bufio"
    "fmt"
    "os"
    "strconv"
    "strings"
)

func main() {
    file, err := os.Open("input.txt")
    if err != nil {
        fmt.Println("Error opening file:", err)
        return
    }
    defer file.Close()

    var number int
    scanner := bufio.NewScanner(file)

    for scanner.Scan() {
        line := scanner.Text()
        if strings.HasPrefix(line, "n=") {
            numStr := strings.TrimPrefix(line, "n=")
            number, _ = strconv.Atoi(numStr)
            if number%2 == 0 {
                number++ // Ensure number is odd
            }
            break
        }
    }

    if err := scanner.Err(); err != nil {
        fmt.Println("Error reading file:", err)
        return
    }

    mid := (number + 1) / 2
    var result strings.Builder

    for i := 1; i <= mid; i++ {
        result.WriteString(strings.Repeat(" ", mid-i))
        result.WriteString(strings.Repeat("*", 2*i-1))
        result.WriteString("\n")
    }

    for i := mid - 1; i >= 1; i-- {
        result.WriteString(strings.Repeat(" ", mid-i))
        result.WriteString(strings.Repeat("*", 2*i-1))
        result.WriteString("\n")
    }

    err = os.WriteFile("output.txt", []byte(result.String()), 0644)
    if err != nil {
        fmt.Println("Error writing file:", err)
    }
}

File: Task3/Java/subtask4.java
import java.io.*;

public class subtask4 {
    public static void main(String[] args) {
        int n = 0;

        try (BufferedReader reader = new BufferedReader(new FileReader("input.txt"))) {
            String line;
            while ((line = reader.readLine()) != null) {
                if (line.startsWith("n=")) {
                    n = Integer.parseInt(line.substring(2).trim());
                    if (n % 2 == 0) n++;
                    break;
                }
            }
        } catch (IOException | NumberFormatException e) {
            System.err.println("Error reading input file.");
            return;
        }

        try (BufferedWriter writer = new BufferedWriter(new FileWriter("output.txt"))) {
            for (int i = 1; i <= n; i += 2) {
                writer.write(" ".repeat((n - i) / 2) + "*".repeat(i));
                writer.newLine();
            }
            for (int i = n - 2; i > 0; i -= 2) {
                writer.write(" ".repeat((n - i) / 2) + "*".repeat(i));
                writer.newLine();
            }
        } catch (IOException e) {
            System.err.println("Error writing to output file.");
        }
    }
}
